bindings/cs/MongoDB.Libmongocrypt/CryptClientFactory.cs

```diff
@@ -23,14 +24,47 @@ namespace MongoDB.Libmongocrypt
     /// </summary>
     public class CryptClientFactory
     {
+        // MUST be static fields since otherwise these callbacks can be collected via the garbage collector
+        // regardless they're used by mongocrypt level or no
+        private static Library.Delegates.HashCallback __cryptoHashCallback = new Library.Delegates.HashCallback(HashCallback.Hash);
+        private static Library.Delegates.CryptoCallback __crypto256EncryptCallback = new Library.Delegates.CryptoCallback(CipherCallbacks.Encrypt);
+        private static Library.Delegates.CryptoCallback __crypto256DecryptCallback = new Library.Delegates.CryptoCallback(CipherCallbacks.Decrypt);
+        private static Library.Delegates.RandomCallback __randomCallback = new Library.Delegates.RandomCallback(SecureRandomCallback.GenerateRandom);
+        private static Library.Delegates.CryptoHmacCallback __cryptoHmacSha512Callback = new Library.Delegates.CryptoHmacCallback(HmacShaCallbacks.HmacSha512);
+        private static Library.Delegates.CryptoHmacCallback __cryptoHmacSha256Callback = new Library.Delegates.CryptoHmacCallback(HmacShaCallbacks.HmacSha256);
+
         /// <summary>Creates a CryptClient with the specified options.</summary>
         /// <param name="options">The options.</param>
         /// <returns>A CryptClient</returns>
         public static CryptClient Create(CryptOptions options)
         {
-            MongoCryptSafeHandle handle = Library.mongocrypt_new();
+            var handle = Library.mongocrypt_new();
+
+            var status = new Status();
+
+            // The below code can be avoided on Windows. So, we don't call it on this system 
```

Comment:
This is good to know that the native crypto in libmongocrypt still provides value